mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAAttachmentMapper.java

```diff
@@ -0,0 +1,152 @@
+        Preconditions.checkArgument(owner != null, "Attachment shouldn't be null when trying to store attachment for owner!");
+        Preconditions.checkArgument(attachment != null, "Username shouldn't be null when trying to store attachment for the owner!");
+        JPAAttachment jpaAttachment = getJpaAttachment(attachment);
+        try {
+            jpaAttachment.getOwners().add(owner.getValue());
+            getEntityManager().getTransaction().begin();
+            getEntityManager().merge(jpaAttachment);
+            getEntityManager().getTransaction().commit();
+        } catch (PersistenceException e) {
+            getEntityManager().getTransaction().rollback();
+            LOGGER.error("Persist attachment" + attachment.getAttachmentId() + " for owner " + owner.getValue() +" met an exception", e);
+            throw new MailboxException(" Store attachment for owner " + owner.getValue() + " failed", e);
+        }
+    }
+
+    @Override
+    public void storeAttachmentsForMessage(Collection<Attachment> attachments, MessageId ownerMessageId) throws MailboxException {
+
+        Preconditions.checkArgument(attachments != null, "Attachment collection shouldn't be null when trying to store attachments for message!");
+        Preconditions.checkArgument(ownerMessageId != null, "MessageId shouldn't be null when trying to store attachments for the message!");
+        try {
+            attachments.stream()
+                    .map(attachment -> getJpaAttachment(attachment))
+                    .forEach(jpaAttachment -> {
+                        jpaAttachment.getMessageIds().add(ownerMessageId.serialize());
+                        getEntityManager().getTransaction().begin();
+                        getEntityManager().merge(jpaAttachment);
+                        getEntityManager().getTransaction().commit();
+                    });
+        } catch (PersistenceException e) {
+            getEntityManager().getTransaction().rollback();
+            LOGGER.error("Persist attachments for message" + ownerMessageId.serialize() + " met an exception ", e);
+            throw new MailboxException("Store attachments for owner with message " + ownerMessageId.serialize() +" failed.", e);
+        }
+
+
+    }
+
+    @Override
+    public Collection<MessageId> getRelatedMessageIds(AttachmentId attachmentId) throws MailboxException {
+        Preconditions.checkArgument(attachmentId != null, "AttachmentId can't be null when trying to get related message Ids!");
+        JPAAttachment attachment = getEntityManager().find(JPAAttachment.class, attachmentId.getId());
+        if(attachment != null && attachment.getMessageIds() != null) {
+            return attachment.getMessageIds()
+                    .stream()
+                    .map(messageId -> new JPAMessageId.Factory().fromString(messageId))
+                    .collect(Guavate.toImmutableList());
+        } else {
+            return new ArrayList<>();
+        }
+
```

Comment:
I don't know how to use Optional in this use case for if the both are null a new arraylist of different type than JPAAttachment will be returned.